### PR TITLE
robot.e.registerIntegration(), robot.e.create() and  auth interfaces phase 0

### PR DESCRIPTION
**Pull request by mosheBelostotsky**
`robot.e.registerIntegration()`, `robot.e.create()` and `auth` proposal for this phase

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/eedevops/he-design/pull/9" target="_blank">ReviewNinja

File: README.md [hunk 1]
```diff
@@ -67,23 +67,136 @@ The `hubot-enterprise` refers to an *enhanced* version of `hubot` that is compri
...
+             short_desc: "...",
+             long_desc: "..."
+           },
+           auth: {
+             type: "...",
+             options: {callback: function}
+           },
+           verbs: [
+             {
+               // robot.e.create object
```
Comment (mosheBelostotsky) [inline README.md]: at this point to be as much similar to existing `robot.e.help`

File: README.md [hunk 2]
```diff
@@ -67,23 +67,136 @@ The `hubot-enterprise` refers to an *enhanced* version of `hubot` that is compri
...
+
+  ```
+
+    **API**
+
+      - `data`: function data.
+        - `product`: will override original product name.
+        - `verb`: single word.
+        - `entity`: optional single word.
+        - `extra`: optional string, this part may be matched and accessed later using msg.match[], depends on the provided regex.
```
Comment (mosheBelostotsky) [inline README.md]: Need to close the final functionality RE:
eedevops/hubot-enterprise#39 and eedevops/hubot-enterprise#40

File: README.md [hunk 3]
```diff
@@ -67,23 +67,136 @@ The `hubot-enterprise` refers to an *enhanced* version of `hubot` that is compri
...
+        if registrar[integration].auth && info.auth
+        # pass auth to auth engine (use bluebird promises)
+        return auth_engine.check_auth(userInfo, integration)
+        .then (res) ->
+          # add secrets to msg object
+          msg.auth_secrets = res
+          callback(msg)
+        .catch (err) ->
+          if (e == 'need_auth')
+            msg.reply "This call need auth, please wait for auth"
```
Comment (mosheBelostotsky) [inline README.md]: Will get this message from messages object.
Later it may be exported to JSON file for multi-language support

**Comment by rqc** [comment]
I submitted minor changes, merging.

**Pull request merged**